### PR TITLE
Audit fixes

### PR DESCRIPTION
**Pull request by kiseln**
https://github.com/AuditOneAuditReviews/Omnibridge2-audit-review/issues/3
https://github.com/AuditOneAuditReviews/Omnibridge2-audit-review/issues/24
https://github.com/AuditOneAuditReviews/Omnibridge2-audit-review/issues/10
https://github.com/AuditOneAuditReviews/Omnibridge2-audit-review/issues/19

Plus, validating `msg.value` on the EVM side

### CODE REVIEW

**Review by olga24912 — APPROVED** [review]

File: evm/src/omni-bridge/contracts/OmniBridge.sol [hunk 1]
```diff
@@ -229,8 +230,11 @@ contract OmniBridge is
             revert InvalidFee();
         }
 
-        uint256 extensionValue;
+        if (msg.value != valueRequired(tokenAddress, amount, nativeFee)) {
+            revert InvalidValue();
+        }
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply evm/src/omni-bridge/contracts/OmniBridge.sol:235]: Wait, what about regular `fee`? I thought that we need to check it as well, but `valueRequired` doesn't consider it. Also, for Sepolia chains `messageFee` is `0`, but for other chains user might pay for it and then `msg.value` would be greater than a value calculated by `valueRequired`

[the remark this answers is not in the document]
Comment (olga24912) [reply evm/src/omni-bridge/contracts/OmniBridge.sol:235]: fee refers to tokens, never Ether. valueRequired is a virtual function, and it’s calculated correctly.

[the remark this answers is not in the document]
Comment (kiseln) [reply evm/src/omni-bridge/contracts/OmniBridge.sol:235]: Ah so for wormhole it is already checked when sending a message? Then I'd probably just check for Ethereum version like you say

[the remark this answers is not in the document]
Comment (kiseln) [reply evm/src/omni-bridge/contracts/OmniBridge.sol:240]: Let's keep this perhaps in order not to rely on wormhole

**Review by karim-en — CHANGES_REQUESTED** [review]

File: evm/src/omni-bridge/contracts/OmniBridge.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (karim-en) [reply evm/src/omni-bridge/contracts/OmniBridge.sol]: aa36c0049f6979538dd48f34d4e537e75b4925bd

**Review by karim-en — APPROVED** [review]

**Review by olga24912 — APPROVED** [review]

**Pull request merged**